### ISSUE TITLE
OS-specific path separator problem

### ISSUE THREAD

**Issue by xiongjiwei** [beyondstorage/go-service-ftp#22]
https://github.com/beyondstorage/go-service-ftp/blob/f5d7dba93642a84ee9be79227167c6bed154aa25/utils.go#L139-L142

`filepath.Separator` is an OS-specific path separator, on windows, it is `\`, on UNIX it is `/`. it will cause the path problem, if some write a file on windows, the path may look like `\path/to/file`, and the read it on UNIX, the path is `\path\to\file`, I do not know it is the server or client responsibility, should the server handle this two style path or client provide the correct path according to the `syst` command?

**Comment by xiongjiwei** [comment]
also, it here should be 
```go
if !strings.HasSuffix(path, string(filepath.Separator)) { 
 	absPath += string(filepath.Separator) 
}
```

**Comment by Xuanwo** [comment]
It's a wide problem exists in all our services. Our original design requires user input UNIX style path. But it proved to be impossible. We will start a new GSP in which we will make the path handle logic clear.

**Comment by Xuanwo** [comment]
This issue should be resolved by https://github.com/beyondstorage/go-storage/issues/649

**Comment by Xuanwo** [comment]
We are working on resolving this issue via new GSP: https://github.com/beyondstorage/go-storage/pull/749

**Comment by Xuanwo** [comment]
Already fixed in GSP-749: beyondstorage/go-storage#749
